contracts/NestedFactory.sol

```diff
@@ -273,39 +288,34 @@ contract NestedFactory is INestedFactory, ReentrancyGuard, Ownable, MixinOperato
     /// @dev For every orders, call the operator with the calldata
     /// to submit buy orders (where the input is one asset).
     /// @param _nftId The id of the NFT impacted by the orders
-    /// @param _inputToken Token used to make the orders
-    /// @param _inputTokenAmount Amount of input tokens to use
-    /// @param _orders Orders calldata
+    /// @param _batch The order to process
     /// @param _reserved True if the output is store in the reserve/records, false if not.
     /// @param _fromReserve True if the input tokens are from the reserve
     /// @return feesAmount The total amount of fees
     /// @return tokenSold The ERC20 token sold (in case of ETH to WETH)
     function _submitInOrders(
         uint256 _nftId,
-        IERC20 _inputToken,
-        uint256 _inputTokenAmount,
-        Order[] calldata _orders,
+        BatchedOrder calldata _batch,
         bool _reserved,
         bool _fromReserve
     ) private returns (uint256 feesAmount, IERC20 tokenSold) {
-        _inputToken = _transferInputTokens(_nftId, _inputToken, _inputTokenAmount, _fromReserve);
+        require(_batch.orders.length != 0, "NF: INVALID_ORDERS");
```

Comment:
Not needed :) 
The `_transferInputTokens` will fail if the user is using a wrong token or a wrong amount.
He can't transfer more than what he has (with the `transferFrom`), it will also fail with amount `0`.